### PR TITLE
1458 user profile page

### PR DESCRIPTION
**Pull request by cy**
screencapture-localhost-3000-user-profile-1503608081412

### CODE REVIEW

**Comment by alex-wilmer** [comment]
this breaks if you go to that page not logged in. maybe add a branch + redirect?

```
const enhance = compose(
  connect(({ auth: { user } }) => ({ user })),
  branch(({ user }) => !user, renderComponent(() => <Redirect to="/" />)),
);
```

Also could you please create a folder in `routes` with your Route + Page files? I'm slowly clearing out the `containers` folder which was meant for classic relay containers. thanks!

File: src/packages/@ncigdc/routes/UserProfileRoute/UserProfileRoute.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import React from 'react';
+import { compose, branch, renderComponent } from 'recompose';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router';
+
+import { Row } from '@ncigdc/uikit/Flex';
+import Hidden from '@ncigdc/components/Hidden';
+import Heading from '@ncigdc/uikit/Heading';
+import Table, { Tr, Td } from '@ncigdc/uikit/Table';
+import CheckedSquareO from '@ncigdc/theme/icons/CheckedSquareO';
```
Comment (Jephuff) [inline src/packages/@ncigdc/routes/UserProfileRoute/UserProfileRoute.js]: How would you feel about just a checkmark without the square, and nothing if they don't have it? These make me feel like I can change my permissions haha

Comment (cy) [reply src/packages/@ncigdc/routes/UserProfileRoute/UserProfileRoute.js]: haha true

**Comment by cy** [comment]
screencapture-localhost-3000-user-profile-1503688743861

**Comment by alex-wilmer** [comment]
what do you think of using fullwidth layout title for username

image

```
    <FullWidthLayout
      title={'ALEX WILMER'}
      entityType={<i className="fa fa-user" />}
    >
```

**Comment by cy** [comment]
Sure - kinda looks like the case entity page though, guess the icon is 'CA' there. Currently fighting with travis

**Comment by cy** [comment]
screen shot 2017-08-28 at 2 06 18 pm
as a modal. ready for re-review

File: src/packages/@ncigdc/components/UserDropdown.js [hunk 2]
```diff
@@ -50,6 +59,12 @@ const UserDropdown = connect(state => ({
         </NavLink>
       }
     >
+      <DropdownItem onClick={() => dispatch(setModal(<UserProfileModal />))}>
```
Comment (Jephuff) [inline src/packages/@ncigdc/components/UserDropdown.js]: Could you add `cursor: pointer` to these?

File: src/packages/@ncigdc/components/Modals/UserProfileModal.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+              }}
+            >
+              <SearchIcon size={14} />
+            </div>
+            <Hidden>filter cases</Hidden>
+          </label>
+          <input
```
Comment (Jephuff) [inline src/packages/@ncigdc/components/Modals/UserProfileModal.js]: can you either pass in `searchTerm` as the initial value or reset the query param on open? 
right now if you search for something and reopen the modal the results are filters but the input is empty.

File: src/packages/@ncigdc/components/Modals/UserProfileModal.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+          />
+        </Row>
+        <Row style={{ justifyContent: 'center' }}>
+          <LocalPaginationTable
+            className="test-consequences-table"
+            data={data}
+            prefix={paginationPrefix}
+            entityName="projects"
+            style={{ width: '70%' }}
+            enablePickSize={false}
```
Comment (Jephuff) [inline src/packages/@ncigdc/components/Modals/UserProfileModal.js]: Do we need to disable picking size? 10 is a pretty small number, might be better to just have overflow?

Comment (cy) [reply src/packages/@ncigdc/components/Modals/UserProfileModal.js]: both scrolling and pagination in a modal seems too much and also didn't want to deal with small issues like the picker being under the modal
screen shot 2017-08-30 at 3 17 43 pm
maybe I'll do away with the front end pagination and see what phuong-my thinks of just scroll

Comment (shanewilson) [reply src/packages/@ncigdc/components/Modals/UserProfileModal.js]: the full page you had before seems better

Comment (cy) [reply src/packages/@ncigdc/components/Modals/UserProfileModal.js]: reason for changing to modal was they didn't want it to be a full data page, and they wanted to "have a screen where users can see it without navigating away from current screen"

**Comment by cy** [comment]
@alex-wilmer @Jephuff ready for re-veiw

**Pull request merged**